api.js

```diff
@@ -0,0 +1,131 @@
+const server = require('./server');
+const db = require('./data/db');
+
+
+//This is my API file, I have it separated from the server.js file so that I can keep my code cleaner and simplify my calls.
+
+
+// Named methods on db.js
+// find,
+// findById,
+// insert,
+// update,
+// remove.
+
+
+
+
+class API {
+
+    constructor({ url }){
+    this.url = "/api/users"
+    this.endpoints = {}
+    }
+    // Create and store a single entity's endpoints
+    //@param {A user Object} user
+    createUser(user) {
+    this.endpoints[user.id] = this.createBasicCRUDEndpoints(user)
+    }
+
+    createUsers(users) {
+    users.forEach(this.createUser.bind(this))
+    }
+
+    //Create the basic endpoints handlers for CRUD operations
+    // @param {A user Object} user
+
+    createBasicCRUDEndpoints( {id} ) {
+    var endpoints = {}
```

Comment:
Do not use var, use let, but in this case I think const works, since you are only editing the properties of the object.